adapters/audienceNetwork/facebook.go

```diff
@@ -359,6 +337,8 @@ func (this *FacebookAdapter) MakeBids(request *openrtb.BidRequest, adapterReques
 
 	for _, seatbid := range bidResp.SeatBid {
 		for _, bid := range seatbid.Bid {
+			bid := bid
```

Comment:
What's the reason for assigning `bid` to `bid` here? This should have no effect, right?

Reply:
It has an affect. This fixes a memory issue with passing a reference to a loop iterator variable outside the loop. Contributors have pointed out the issue in recent PRs. The new adapter docs now include the corrected code and many adapters have been updated, but we should do a review to ensure they're all fixed.

The root issue is passing a reference to the `bid` loop variable outside of the loop. The `bid` variable is re-used with each loop iteration and is overwritten with a shallow copy of the current item. The fix is to either create a new shallow copy or to use the index and access the bid item directly from the slice. This is the former approach. Since Go allows for variable shadowing, this line is creating a new variable `bid` as a shallow copy of the loop iterator variable `bid`.  An alternative is to avoid using `range` here altogether.

Reply:
Rewritten to use index instead of shadow copying the bid loop variable.